storage2/src/snapshot.rs

```diff
@@ -9,38 +9,38 @@ use tracing::Span;
 
 use crate::state::StateRead;
 
-/// Snapshots maintain a point-in-time view of the underlying storage, suitable
-/// for read-only access by multiple threads, i.e. RPC calls.
+mod rocks_wrapper;
+use rocks_wrapper::RocksDbSnapshot;
+
+/// A snapshot of the underlying storage at a specific state version, suitable
+/// for read-only access by multiple threads, e.g., RPC calls.
 ///
-/// This is implemented as a wrapper around a [RocksDB snapshot](https://github.com/facebook/rocksdb/wiki/Snapshot)
-/// with an associated JMT version number for the snapshot.
+/// Snapshots are cheap to create and clone.  Internally, they're implemented as
+/// a wrapper around a [RocksDB
+/// snapshot](https://github.com/facebook/rocksdb/wiki/Snapshot) with a pinned
+/// JMT version number for the snapshot.
 #[derive(Clone)]
-pub(crate) struct Snapshot(pub(crate) Inner);
+pub struct Snapshot(Arc<Inner>);
```

Comment:
Moving to an internal refcount simplifies the implementation, because when using `spawn_blocking`, we effectively need a way to access the same `Snapshot` from multiple threads.  The `Arc<Inner>` pattern means we just have to increment a refcount to pass a copy of ourself to the task we spawn.